src/partest/scala/tools/partest/nest/Runner.scala

```diff
@@ -533,17 +548,18 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
     // pass if it checks and didn't crash the compiler
     // or, OK, we'll let you crash the compiler with a FatalError if you supply a check file
     def checked(r: CompileRound) = r.result match {
+      case s: Skip => s
       case crash @ Crash(_, t, _) if !checkFile.canRead || !t.isInstanceOf[FatalError] => crash
-      case dnc @ _ => diffIsOk
+      case _ => diffIsOk
     }
 
-    compilationRounds(testFile).find(!_.result.isOk).map(checked).getOrElse(genFail("expected compilation failure"))
+    compilationRounds(testFile).find(r => !r.result.isOk || r.result.isSkipped).map(checked).getOrElse(genFail("expected compilation failure"))
```

Comment:
Always sad to lose an obscure use of underscore.

Previously, tests could not declare themselves skipped, but could only succeed trivially? There is "isJavaAtLeast" somewhere.